pandas/_libs/window/indexers.pyx

```diff
@@ -10,64 +10,122 @@ from numpy cimport ndarray, int64_t
 # These define start/end indexers to compute offsets
 
 
-class FixedWindowIndexer:
-    """
-    create a fixed length window indexer object
-    that has start & end, that point to offsets in
-    the index object; these are defined based on the win
-    arguments
-
-    Parameters
-    ----------
-    values: ndarray
-        values data array
-    win: int64_t
-        window size
-    index: object
-        index of the values
-    closed: string
-        closed behavior
-    """
-    def __init__(self, ndarray values, int64_t win, object closed, object index=None):
+class BaseIndexer:
+    """Base class for window bounds calculations"""
+
+    def __init__(
+        self,
+        **kwargs,
+    ):
+        """
+        Parameters
+        ----------
+        **kwargs :
+            keyword argument that will be available when get_window_bounds is called
+        """
+        self.__dict__.update(kwargs)
+
+    def get_window_bounds(
+        self,
+        num_values: int = 0,
+        window_size: int = 0,
+        min_periods: Optional[int] = None,
```

Comment:
can/should we use -1 to indicate no limit for consistency with the groupby code?

Reply:
Sure. Since it's a new API happy to unify it with existing conventions.